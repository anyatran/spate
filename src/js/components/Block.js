import React from 'react'
import PropTypes from 'prop-types'
import BlockContent from './BlockContent'

const Block = ({ bg, details, subtitle, title, url }) => (
  <div className="block">
    {url &&
      <a href={url} className="block__container block__container--link" style={bg && {backgroundImage: `url(${bg})`}}>
        <BlockContent title={title} subtitle={subtitle} details={details}/>
      </a>
    }
    {!url &&
      <div className="block__container" style={bg && {backgroundImage: `url(${bg})`}}>
        <BlockContent title={title} subtitle={subtitle} details={details}/>
      </div>
    }
  </div>
)

Block.propTypes = {
  bg: PropTypes.string,
  details: PropTypes.string,
  subtitle: PropTypes.string,
  title: PropTypes.string,
  url: PropTypes.string
}

export default Block
