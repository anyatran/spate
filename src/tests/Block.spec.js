import React from 'react'
import { mount } from 'enzyme'
import Block from '../js/components/Block'

describe('<Block />', () => {
  it('should render Block', () => {
    const wrapper = mount(<Block title="block title" subtitle="block subtitle"/>)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render Block title', () => {
    const wrapper = mount(<Block title="block title"/>)
    expect(wrapper.find('.block__title').text()).toEqual('block title')
    expect(wrapper.find('.block__subtitle').length).toEqual(0)
    expect(wrapper.find('.block__details').length).toEqual(0)
  })

  it('should render Block subtitle', () => {
    const wrapper = mount(<Block subtitle="block subtitle"/>)
    expect(wrapper.find('.block__title').length).toEqual(0)
    expect(wrapper.find('.block__subtitle').text()).toEqual('block subtitle')
    expect(wrapper.find('.block__details').length).toEqual(0)
  })

  it('should render Block details', () => {
    const wrapper = mount(<Block details="block details"/>)
    expect(wrapper.find('.block__title').length).toEqual(0)
    expect(wrapper.find('.block__details').text()).toEqual('block details')
    expect(wrapper.find('.block__subtitle').length).toEqual(0)
  })

  it('should render Block link', () => {
    const wrapper = mount(<Block details="block details" url="some/url"/>)
    expect(wrapper.find('a').length).toEqual(1)
  })
})
