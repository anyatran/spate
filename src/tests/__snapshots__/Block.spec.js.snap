// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Block /> should render Block 1`] = `
<Block
  subtitle="block subtitle"
  title="block title"
>
  <div
    className="block"
  >
    <div
      className="block__container"
    >
      <BlockContent
        subtitle="block subtitle"
        title="block title"
      >
        <div>
          <p
            className="block__title"
          >
            block title
          </p>
          <p
            className="block__subtitle"
          >
            block subtitle
          </p>
        </div>
      </BlockContent>
    </div>
  </div>
</Block>
`;
